lib/money/money.rb

```diff
@@ -390,11 +415,27 @@ def symbol
     currency.symbol || "¤"
   end
 
+  def infinite_precision?
+    @infinite_precision
+  end
+
   # Common inspect function
   #
   # @return [String]
   def inspect
-    "#<#{self.class.name} fractional:#{fractional} currency:#{currency}>"
+    "#<#{self.class.name} fractional:#{fractional} currency:#{currency}"
+  end
+
+  def to_precise
+    return self if infinite_precision?
+
+    self.class.new(fractional, currency, { bank: bank, infinite_precision: true })
+  end
+
+  def to_rounding
```

Comment:
Any other name suggestions for the conversion methods? It's definitely not immediately obvious what these are for

Reply:
Hmm 🤔, yeah `to_rounding` is particularly unclear.  Perhaps `to_low_precision` / `to_high_precision`?  Which reminds me that the term "infinite_precision" also feels a bit weird... it is indeed very high, but not infinite... though it could be said the money gem's precision is infinite and only limited by BigDecimal.

Also, perhaps we don't need these convenience methods just yet?  They are theoretically nice, but they are not used anywhere but their corresponding specs right now.  Our team even made corresponding modifications in `monetize` and `money-rails` for per instance support without needing them.  If future use cases arise, perhaps that would inform the naming.  That said, @FunkyloverOne may have had a specific use case in mind.

Reply:
Hey @marcusmateus, of course, I had, that use case was exactly why I've started this PR :grinning: 
```ruby
    it "allows to avoid rounding loss errors" do
      expect(Money.new(100, 'NZD') / 3 * 100).to eq(Money.new(3300, 'NZD'))

      result = (Money.new(100, 'NZD').to_precise / 3 * 100).to_rounding
      expect(result).to eq(Money.new(3333, 'NZD'))
    end
```
Notice the difference in 33 cents in this particular case. This change allows performing more _precise_ calculations on money objects, while using their _"rounding"_ form everywhere else.

And, I wanted the name to be as short as possible because it is supposed to be used in "quick" calculations, where you have some fractional multiplying or dividing involved.

Reply:
If this example is the primary intended use, wouldn't it be more convenient to do something like:

```
result = Money.with_infinite_precision do
  Money.new(100, 'NZD' / 3 * 100)
end
```

Or perform even longer computations within the block without having to go back and forth between different precision settings?

I'm also wondering if `infinite_precision` should just be `rounding_mode = nil`.

Sorry for a slow progress on this one, it does change the API and would need to go in a major release

Reply:
@antstorm that blocky version could also be the case, but it still would be helpful to convert back to "rounding" after all the computations at some point. That's where those helpers come in handy.

> I'm also wondering if infinite_precision should just be rounding_mode = nil.

I would say that's a bit different concepts because we still want to know which mode to use when we perform any rounding, even when we call `#round` explicitly. Also `rounding_mode` is a class-level setting, we do not currently set it on instance level.
Or maybe I just misunderstood you.

Reply:
@antstorm I have a question about your "blocky" approach: would it be thread-safe? :thinking: 
Also, would it be able to affect money instances initialized out of the block (before)? Like this:
```ruby
m = Money.new(100, 'NZD')
result = Money.with_infinite_precision { m / 3 * 100 }
```
Anyway, I guess it could be added in a separate PR.

---

Now, speaking of my naming: doesn't `to_rounding` actually make sense? It produces an automatically rounding instance. And right now - all of them are implicitly automatically rounding by default (unless you set `infinite_precision` to `true` for all of them).
I think that `to_precise` is actually a bit confusing instead, as what it does - it produces an instance that does not perform automatic rounding. So it could also be simply called `to_non_rounding`. It seems clear to me.
Real money is usually rounded to its subunit. So does ruby money right now.

I think that `to_precise` and `to_non_rounding` could be aliases. `to_precise` is shorter and seems clear enough.